airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry_entry.py

```diff
@@ -0,0 +1,380 @@
+
+    return registry_entry, connector_type
+
+
+def get_connector_type_from_registry_entry(registry_entry: dict) -> TaggedRegistryEntry:
+    if registry_entry.get("sourceDefinitionId"):
+        return ("source", ConnectorRegistrySourceDefinition)
+    elif registry_entry.get("destinationDefinitionId"):
+        return ("destination", ConnectorRegistryDestinationDefinition)
+    else:
+        raise Exception("Could not determine connector type from registry entry")
+
+
+def get_registry_entry_write_path(metadata_entry: LatestMetadataEntry, registry_name: str):
+    metadata_path = metadata_entry.file_path
+    if metadata_path is None:
+        raise Exception(f"Metadata entry {metadata_entry} does not have a file path")
+
+    metadata_folder = os.path.dirname(metadata_path)
+    print(f"metadata_folder: {metadata_folder}")
+    return os.path.join(metadata_folder, registry_name)
+
+
+def persist_registry_entry_to_json(
+    registry_entry: PolymorphicRegistryEntry,
+    registry_name: str,
+    metadata_entry: LatestMetadataEntry,
+    registry_directory_manager: GCSFileManager,
+) -> GCSFileHandle:
+    """Persist the registry_entry to a json file on GCS bucket
+
+    Args:
+        registry_entry (ConnectorRegistryV0): The registry_entry.
+        registry_name (str): The name of the registry_entry. One of "cloud" or "oss".
+        metadata_entry (LatestMetadataEntry): The related Metadata Entry.
+        registry_directory_manager (OutputDataFrame): The registry_entry directory manager.
+
+    Returns:
+        OutputDataFrame: The registry_entry directory manager.
+    """
+    registry_entry_write_path = get_registry_entry_write_path(metadata_entry, registry_name)
+    registry_entry_json = registry_entry.json(exclude_none=True)
+    file_handle = registry_directory_manager.write_data(registry_entry_json.encode("utf-8"), ext="json", key=registry_entry_write_path)
+    return file_handle
+
+
+def generate_and_persist_registry_entry(
+    metadata_entry: LatestMetadataEntry,
+    cached_specs: OutputDataFrame,
+    metadata_directory_manager: GCSFileManager,
+    registry_name: str,
+) -> str:
+    """Generate the selected registry from the metadata files, and persist it to GCS.
+
+    Args:
+        context (OpExecutionContext): The execution context.
+        metadata_entry (List[LatestMetadataEntry]): The metadata definitions.
+        cached_specs (OutputDataFrame): The cached specs.
+
+    Returns:
+        Output[ConnectorRegistryV0]: The registry.
+    """
+    raw_entry_dict = metadata_to_registry_entry(metadata_entry, registry_name)
+    registry_entry_with_spec = apply_spec_to_registry_entry(raw_entry_dict, cached_specs)
+
+    _, ConnectorModel = get_connector_type_from_registry_entry(registry_entry_with_spec)
+
+    registry_model = ConnectorModel.parse_obj(registry_entry_with_spec)
+
+    file_handle = persist_registry_entry_to_json(registry_model, registry_name, metadata_entry, metadata_directory_manager)
```

Comment:
Does this persist to the versioned folder, or to latest? Or both?
Not sure I undertand how/when we save to e.g. `/1.0.0/oss.json` vs `/latest/oss.json`

Reply:
oh! maybe we're only writing to the versioned one actually, since "latest" would be in the full registry?

Reply:
It writes `oss.json` and/or `cloud.json` to the same folder that `metadata.yaml` was found.

Because we run the `registry_entry` asset against `all_metadata_files` then we write a registry entry to anywhere a metadata file exists.

This includes latest and any versions.